test/Generator/MethodGeneratorTest.php

```diff
@@ -74,6 +74,38 @@ public function testMethodParameterMutator()
         $methodGenerator->setParameter(new stdClass());
     }
 
+    public function testMethodParameterSetPosition()
+    {
+        $methodGenerator = new MethodGenerator();
+
+        $methodGenerator->setParameter('foo');
+        $methodGenerator->setParameter(['name' => 'bar', 'type' => 'array', 'position' => 2]);
+        $methodGenerator->setParameter(
+            ParameterGenerator::fromArray(['name' => 'baz', 'type' => stdClass::class, 'position' => 1])
+        );
+
+        $params = $methodGenerator->getParameters();
+        self::assertCount(3, $params);
+
+        /** @var ParameterGenerator $foo */
+        $foo = array_shift($params);
```

Comment:
Suggestion:

 1. `$sorting = array_map(static function (ParameterGenerator $parameter): string { return $parameter->getName(); }, $params);`
 2. `self::assertEquals(['foo', 'bar', 'baz'], $sorting);`

Would make the test much simpler to read, IMO, and removes the need for counting returned values.